OrbitLinuxTracing/PerfEventOpen.cpp

```diff
@@ -77,6 +77,20 @@ int sample_event_open(uint64_t period_ns, pid_t pid, int32_t cpu) {
   return generic_event_open(&pe, pid, cpu);
 }
 
+int sample_event_fp_open(uint64_t period_ns, pid_t pid, int32_t cpu) {
+  perf_event_attr pe = generic_event_attr();
+  pe.type = PERF_TYPE_SOFTWARE;
+  pe.config = PERF_COUNT_SW_CPU_CLOCK;
+  pe.sample_period = period_ns;
+  pe.sample_type |= PERF_SAMPLE_CALLCHAIN;
+  // TODO(kuebler): This shouldn't be a random constant
```

Comment:
Also `sample_max_stack` shouldn't be larger than the maximum specified in `/proc/sys/kernel/perf_event_max_stack`. You could read that value (by default it's 127).

Reply:
I'll update it to 127 and have a todo there to read it from /proc/sys/kernel/perf_event_max_stack.